Hannah.gd

```diff
@@ -3,34 +3,47 @@ extends CharacterBody2D
 
 var speed = 300.0
 
+
+@onready var subviewport = $"../../../UI/PlayerInventoryUI/InvSprite/SubViewportContainer/SubViewport"
+
 #Get the inventory object - probably this should just be a child of the player? TODO refactor
 @onready var inventory = get_tree().get_nodes_in_group("Inventory")[0]
 var equipped
 var interacting = false
+var child_count_with_equipped 
+var startInvVisible = true
+
+func _ready():
+	child_count_with_equipped = get_child_count() + 1
+	subviewport.world_2d = get_viewport().world_2d
 
 func _input(event):
 	if !interacting:
+		
 		#Here is inventory handling and interacting with world objects
 		if !inventory.visible and event.is_action_pressed("interact") and $RayCast2D.is_colliding() and $RayCast2D.get_collider().has_method("interact"):
 			interacting = true
 			$RayCast2D.get_collider().interact()
-		elif inventory.visible and event.is_action_pressed("interact"):
-			inventory.toggle_context_menu_for_selected()
 		elif event.is_action_pressed("inventory"):
-			if inventory.visible:
+			
+			#Just in case I have the inventory visible at game runtime
+			if inventory.visible and !startInvVisible:
```

Comment:
Add a task to eventually remove this. When there's nothing else of value to be done

Reply:
https://github.com/orgs/StarkandcoEnterprises/projects/2/views/1?pane=issue&itemId=45624071